aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3422,16 +3427,40 @@ static void linalg_lstsq_out_info(
     TORCH_INTERNAL_ASSERT(singular_values.sizes().equals(singular_values_shape));
     TORCH_INTERNAL_ASSERT(singular_values.is_contiguous());
   }
-
   // 'input' is modified in-place so we need a column-major copy
   auto input_working_copy = copyBatchedColumnMajor(input);
-
   // now the actual call that computes the result in-place (apply_lstsq)
-  lstsq_stub(input.device().type(), input_working_copy, solution, rank, singular_values, infos, rcond, driver);
-
+  if(driver == "gelsd" || driver == "gelss") {
+    auto k = std::min(m, n);
+    auto U = at::zeros({input.size(0), m, k}, input.options());
+    auto Vh = at::zeros({input.size(0), k, n}, input.options());
+
+    svd_stub(input.device().type(),
+          input,
+          false, // you don't want the full SVD for least squares
+          true, // we need U, S, Vh for least squares
+          "gesvd",
+          U, singular_values, Vh, infos);
+
+    auto tol = 1e-5; // what should this be? can rcond be used?
```

Comment:
Like the comment says, please lmk if there is a suitable tolerance to be used.

Reply:
In `numpy`, they now use the following:
> machine precision times ``max(M, N)`` where M and N are the input matrix dimensions
https://numpy.org/doc/stable/reference/generated/numpy.linalg.lstsq.html

Reply:
You should use `rcond` here.